ports/stm32/qspi.c

```diff
@@ -74,11 +74,33 @@ static inline void qspi_mpu_enable_mapped(void) {
     // for the memory-mapped region, so 3 MPU regions are used to disable access
     // to everything except the valid address space, using holes in the bottom
     // of the regions and nesting them.
-    // At the moment this is hard-coded to 2MiB of QSPI address space.
     uint32_t irq_state = mpu_config_start();
+    #if (MICROPY_HW_QSPI_MPU_REGION_SIZE > 128)
```

Comment:
I don't see this new configuration `MICROPY_HW_QSPI_MPU_REGION_SIZE` set anywhere?  Shouldn't it at least be set to 2 for PYBD-SFx boards?

Also, for consistency with outer code, these `#if`'s don't need to have parenthesis around the conditional.

Reply:
>  Shouldn't it at least be set to 2 for PYBD-SFx boards?

Yes, how about now? see the update, if it's not defined the region will be the same as the defined flash size.

Reply:
It will need to use `MICROPY_HW_QSPIFLASH_SIZE_BITS_LOG2`

Reply:
I used SIZE_BITS like

```#define MICROPY_HW_QSPI_MPU_REGION_SIZE (MICROPY_HW_SPIFLASH_SIZE_BITS / (8 * 1024 * 1024))```

Or I can just defined it for the boards the use qspi.c

Reply:
I'm pretty sure you need something like:
```c
#define MICROPY_HW_QSPI_MPU_REGION_SIZE (1 << (MICROPY_HW_QSPIFLASH_SIZE_BITS_LOG2 - 3))
```
Because `MICROPY_HW_QSPIFLASH_SIZE_BITS_LOG2` is the setting used for the size of the external QSPI-connected flasg (`MICROPY_HW_SPIFLASH_SIZE_BITS` is different).

Reply:
`MICROPY_HW_SPIFLASH_SIZE_BITS` is the size of the flash in Mbits, for example for PYBD_SF2 it's:

```c
#define MICROPY_HW_SPIFLASH_SIZE_BITS (16 * 1024 * 1024)
```

It's used for the number of blocks:

```c
(op) == BDEV_IOCTL_NUM_BLOCKS ? (MICROPY_HW_SPIFLASH_SIZE_BITS / 8 / FLASH_BLOCK_SIZE) : \
```

This is the same for all boards, are you sure you want to use BITS_LOG2 ?

Reply:
Okay, switched to LOG2

Reply:
> STM32F769DISC only uses MICROPY_HW_QSPIFLASH_SIZE_BITS_LOG2

How did that work before this patch ?

Reply:
> How did that work before this patch ?

Good question!  I guess it would only allow access to the first 2MiB when in XIP mode.  But note that the default build of this board uses the external flash for storage, and in this case XIP mode is *disabled* while storage reads/writes occur.